zerver/actions/streams.py

```diff
@@ -526,6 +527,10 @@ def bulk_add_subscriptions(
                 if sub.active:
                     already_subscribed.append(sub_info)
                 else:
+                    if not stream.push_notifications_enabled:
+                        sub_info.sub.push_notifications = None
+                    else:
+                        sub_info.sub.push_notifications = True
```

Comment:
I am not sure whether we also want `RealmAuditLog` objects for change in `push_notifications`. Let us not make any changes now and wait for feedback from Tim.